tests/main/user-session-env/task.yaml

```diff
@@ -54,13 +54,18 @@ execute: |
     su -c 'sh -c "exec env"' -l "$TEST_FISH_USER" > "${TEST_FISH_USER}-child-env"
 
     SNAP_MOUNT_DIR="$(os.paths snap-mount-dir)"
+    LOCAL_PATH=
+    if tests.info is-snapd-pkg-repo; then
+        MOUNT_DIR="$(os.paths snap-mount-dir)"
```

Comment:
I think this is similar to the udev test that is just not measuring anything real. The test should be skipped on distributions where the systemd generators are too old or they give false sense of security where the reality is that it just doesn't work this way. Those helpers do not re-execute.

Reply:
Let's keep this as-is and add this disclaimer next to this if statement:
```
# NOTE: This test doesn't measure what happens with a system using old classic
# package and a new snapd snap. Instead it measures what happens when the classic
# package is updated to contain the code in the tested branch. The test is valid
# but does not give confidence of the behavior during re-execution, as systemd
# generators do not participate in re-execution.
```